### ISSUE TITLE
Unexpected recreation of domain when adding env variables to vercel project

### ISSUE THREAD

**Issue by cungminh2710** [vercel/terraform-provider-vercel#40]
When we added a new environment variable, I saw that the project domain must be replaced. Just want to double check that this is an expected behaviour.

```
Terraform will perform the following actions:

  # vercel_project.web will be updated in-place
  ~ resource "vercel_project" "web" {
      + environment    = [
          + {
              + id     = (known after apply)
              + key    = "NEXT_PUBLIC_APP_CHECK_RECAPTCHAV3"
              + target = [
                  + "production",
                ]
              + value  = (sensitive value)
            }
        ]
      ~ id             = "prj_XYZ" -> (known after apply)
        name           = "web"
        # (5 unchanged attributes hidden)
    }

  # vercel_project_domain.web_dev must be replaced
-/+ resource "vercel_project_domain" "web_dev" {
      ~ id         = "id_XYZ" -> (known after apply)
      ~ project_id = "prj_XYZ" -> (known after apply) # forces replacement
        # (3 unchanged attributes hidden)
    }
```

**Comment by dglsparsons** [comment]
Hey, thanks for raising this. The simple answer is - yeah this is expected at the moment unfortunately. 

The ID field is set to be a computed field. I'm not sure if there is something that can be done lifecycle wise to let terraform know the field doesn't change when the resource is updated though - at the moment terraform sees any update to a project as potentially changing the ID (which is clearly not the case, but could be for other computed fields). 

I'll have a dig and see if there is anything I can do to prevent this, but I'm not 100% sure if this exists.

**Comment by dglsparsons** [comment]
This is released in v0.6.0. Enjoy! 🎊
